pkg/metrics/consume.go

```diff
@@ -0,0 +1,56 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type ConsumerMetrics struct {
```

Comment:
I am not too sure about the naming of having `Sent` and `Written` as two separate metrics. I think we could potentially use the same `*Metric struct` for both consumer and producer. The value of the tenant label could e.g. just stay empty as could the remote for the producer.

Reply:
OK, I can combine them. Do you think it makes sense to have a single struct for both producer and consumer and then have each command only operate on a subset of the metrics?

Reply:
Or use the existing metrics? On second glance, they do seem pretty general and could make sense for producer or consumer mode.

Reply:
I've taken the approach of using the same metrics for both producer and consumer in the latest commit. I added a retryable error metric as well.

Reply:
That looks great :+1:  and we also get the tenant IDs now in the produce part